runtime/jvmti/jvmtiCapability.c

```diff
@@ -378,6 +378,13 @@ jvmtiAddCapabilities(jvmtiEnv* env,
 			 * Set it to 512KB which is default sampling interval as per JEP 331 specification.
 			 */
 			vm->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(currentThread, 512 * 1024);
+			
+			if (jvmtiData->flags & J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED) {
+				rc = JVMTI_ERROR_NOT_AVAILABLE;
+				goto fail;
+			} else {
```

Comment:
The else isn't needed as the if `goto fail` if there's an issue.

Reply:
Fixed, please have another look.